src/sys/unix/kqueue.rs

```diff
@@ -341,9 +341,8 @@ pub mod event {
             (event.flags & libc::EV_EOF) != 0 && event.fflags != 0
     }
 
-    pub fn is_hup(_: &Event) -> bool {
-        // Not supported.
-        false
+    pub fn is_hup(event: &Event) -> bool {
+        event.filter == libc::EVFILT_WRITE && (event.flags & libc::EV_EOF) != 0
```

Comment:
Why restrict it to `EVFILT_WRITE`?  It's relevant for reading, too.

Reply:
We have `is_read_hup` for that, which is the same check but using `EVFILT_READ`.

Reply:
This must be reverted. The problem is that kqueue will fire this event in unexpected scenarios. 

There is an older PR to 0.6 with more info, but IIRC, kqueue will fire this event when some APIs are called in process. For example, if the user calls shutdown(write) this event is triggered even though the remote could send more data. 

To correctly simulate epoll HUP we would need to track state indicating that the socket has previously seen a read hup and only fire HUP when both the read and the write filters have gotten EOF